openpgp/eax/eax.go

```diff
@@ -0,0 +1,169 @@
+// Copyright (C) 2019 ProtonTech AG
+
+// Package eax provides an implementation of the EAX
+// (encrypt-authenticate-translate) mode of operation, as described in
+// Bellare, Rogaway, and Wagner "THE EAX MODE OF OPERATION: A TWO-PASS
+// AUTHENTICATED-ENCRYPTION SCHEME OPTIMIZED FOR SIMPLICITY AND EFFICIENCY."
+// In FSE'04, volume 3017 of LNCS, 2004
+
+package eax
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"golang.org/x/crypto/openpgp/bytesutils"
+	"bytes"
+	"errors"
+)
+
+const (
+	defaultTagSize   = 16
+	defaultNonceSize = 16
+)
+
+type eax struct {
+	block     cipher.Block // Only AES-{128, 192, 256} supported
+	tagSize   int          // At least 12 bytes recommended
+	nonceSize int
+}
+
+// NewEAX returns an EAX instance with AES-{keyLength} and default tag and
+// nonce lengths.
+func NewEAX(key []byte) eax {
+	aesCipher, err := aes.NewCipher(key)
+	if err != nil {
+		panic(err)
+	}
+
+	return eax {
+		block:     aesCipher,
+		tagSize:   defaultTagSize,
+		nonceSize: defaultNonceSize,
+	}
```

Comment:
Similarly, you could merge the contents of this function into `NewEAXWithNonceAndTagSize` and call that here, rather than mutating the returned struct there.

Reply:
I agree